rules.mk

```diff
@@ -37,35 +37,34 @@ export PATH
 endif
 
 # parses tools.go and returns the tool name prefixed with bin/
-toolsBins := $(addprefix bin/,$(shell cat tools.go | grep _ | awk -F'"' '{print $$2}' | xargs -tI % basename %))
```

Comment:
Also in [`1e90fbc` (#198)](https://github.com/tinkerbell/boots/pull/198/commits/1e90fbc8dacd4412ac7b176241fecfafd3e5860f)

The reason to change this one was just to continue simplification. It wasn't as much of a difference as the the recipe change, but its nice to cut down on useless use of cat and grep when we already are using awk.

Also the `-t` to xargs was always printing `basename $pkg/name` even if there was nothing to do, which was confusing and not the norm:
```
make bin/stringer
basename github.com/golang/mock/mockgen 
basename golang.org/x/tools/cmd/goimports 
basename golang.org/x/tools/cmd/stringer 
make: 'bin/stringer' is up to date.
```

Reply:
I prefer simple to understand over simple to do, which is a design principle I've proposed [here](https://github.com/tinkerbell/boots/pull/188). In my opinion, this doesn't feel any easier to understand than the status quo.

Reply:
Don't agree with this, but it's definitely not doing any harm so won't hold up the PR for it.

Reply:
I fail to see how `addprefix, shell, cat, grep, awk, xargs, basename` is simpler to understand than `addprefix, shell, notdir, awk`, that -4/+1 concepts. xargs alone is confusing for many people.

Note that my awk command is also being more stringent with the files it operates on compare to the grep, it could just as easily be `awk /_/ {print $$2}`. I'd be ok with loosening up its matching if that would help.